File: June 2022/June_23.java
class Solution {
    public int scheduleCourse(int[][] courses) {
        Arrays.sort(courses, (a, b) -> a[1] - b[1]);
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        int curr = 0;
        for(var v : courses){
            if(curr + v[0] <= v[1]){
                curr += v[0];
                pq.add(v[0]);
            } else if(!pq.isEmpty() && pq.peek() > v[0]){
                curr -= pq.remove();
                curr += v[0];
                pq.add(v[0]);
            }
        }
        return pq.size();
    }
}

File: June 2022/June_23.cpp
class Solution {
public:
    static bool cmp(vector<int> &a, vector<int> &b){
        return a[1] < b[1];
    }
    
    int scheduleCourse(vector<vector<int>>& courses) {
        sort(courses.begin(), courses.end(), cmp);
        priority_queue<int> pq;
        int curr = 0;
        for(auto &i : courses){
            if(curr + i[0] <= i[1]){
                curr += i[0];
                pq.push(i[0]);
            } else if(!pq.empty() && pq.top() > i[0]){
                curr -= pq.top();
                pq.pop();
                curr += i[0];
                pq.push(i[0]);
            }
        }
        return pq.size();
    }
};
